### PR TITLE
[dashboard scoped filter] Improve scrollbar inside modal

### PR DESCRIPTION
**Pull request by graceguo-supercat**
### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY
A few UI updates based on designer's review:

1. show tab name  and filter name as regular font-weight, not bold. Only show `All filters` and `All charts` bold font.
2. improve scrollbar behavior when content is long:

Before: scroll bar is outside of modal
oXKrPLR0no

After: modal should have fixed height. when content is overflow, should show scroll bar inside modal, and keep modal header and footer in the fixed position.
ZfYLu60NAF

### TEST PLAN
manual test

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### REVIEWERS
@kenchendesign

### CODE REVIEW

**Comment by rusackas** [comment]
@graceguo-supercat This looks like a good improvement to me!  I'm wondering about a couple enhancements, which I can look at in a follow-up PR if you'd like:

• Would it be even better if just the individual tree were to scroll? In this case, that would be just the right side's tree. Optionally, it could even leave the Expand/Collapse/Search pinned to the top. 

• I see the Less file has some significant-looking static values (128px, 64px, ...). I didn't look super closely at these, but if they're making space for other elements (i.e. a header), would it add value to use Less variables, so that we can tweak more easily in the future? I've been meaning to make a pass at the Less files in general looking for this sort of thing, hoping we can use more Less variables for "style guide" kind of things.

**Review by rusackas — APPROVED** [review]
LGTM, aside from a couple small areas of potential enhancement I mentioned in a separate comment.

**Comment by graceguo-supercat** [comment]
> @graceguo-supercat This looks like a good improvement to me! I'm wondering about a couple enhancements, which I can look at in a follow-up PR if you'd like:
> 
> • Would it be even better if just the individual tree were to scroll? In this case, that would be just the right side's tree. Optionally, it could even leave the Expand/Collapse/Search pinned to the top.
> 

Yes! I talked to @kenchendesign and he told me that this is standard interaction behavior for modals actually sidebar and content should scroll separately. 

I added this changed!

File: superset/assets/src/dashboard/stylesheets/filter-scope-selector.less [hunk 1]
```diff
@@ -19,14 +19,40 @@
...
+    .filter-scope-pane {
+      overflow-y: scroll;
+    }
+  }
+
+  .dashboard-modal-actions-container {
+    height: 64px;
+    border-top: 1px solid #ccc;
+    padding: 24px;
+    margin: 0 0  0 -24px;
```
Comment (etr2460) [inline superset/assets/src/dashboard/stylesheets/filter-scope-selector.less]: nit: extra space between the 0s here

File: superset/assets/src/dashboard/stylesheets/filter-scope-selector.less [hunk 2]
```diff
@@ -108,6 +131,10 @@
     }
 
     .react-checkbox-tree {
+      .rct-title .root {
+        font-weight: bold;
```
Comment (etr2460) [inline superset/assets/src/dashboard/stylesheets/filter-scope-selector.less]: i see here we're using `font-weight: bold;` but below it's `font-weight: 700;`. Can we standardize?

**Review by etr2460 — APPROVED** [review]
a couple css nits, otherwise lgtm

File: superset/assets/src/dashboard/stylesheets/filter-scope-selector.less (hunk 2 above)
Comment (graceguo-supercat) [reply superset/assets/src/dashboard/stylesheets/filter-scope-selector.less]: i choose to use `bold`.

File: superset/assets/src/dashboard/stylesheets/filter-scope-selector.less (hunk 1 above)
Comment (graceguo-supercat) [reply superset/assets/src/dashboard/stylesheets/filter-scope-selector.less]: fixed.

File: superset/assets/src/dashboard/stylesheets/filter-scope-selector.less (hunk 2 above)
Comment (rusackas) [reply superset/assets/src/dashboard/stylesheets/filter-scope-selector.less]: Not to sidetrack the discussion, but I'm considering if it's worth moving some of these values (and other related typographic treatments) to LESS variables, so we can use `font-weight: @bold` and set/theme that to be 700 (or whatever!) globally.

**Pull request merged**